managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1498,12 +1545,13 @@ public void asyncDelete(Position pos, final AsyncCallbacks.DeleteCallback callba
 
         // Apply rate limiting to mark-delete operations
         if (markDeleteLimiter != null && !markDeleteLimiter.tryAcquire()) {
+            lastMarkDeleteEntry = new PendingMarkDeleteEntry(newMarkDeletePosition, Collections.emptyMap(), null, null);
             callback.deleteComplete(ctx);
             return;
         }
 
         try {
-            internalAsyncMarkDelete(newMarkDeletePosition, new MarkDeleteCallback() {
+            internalAsyncMarkDelete(newMarkDeletePosition, Collections.emptyMap(), new MarkDeleteCallback() {
```

Comment:
we are creating map even in case we don't need to store properties for cursor. Can we store it as `null/Optional` to avoid additional object creation (yes, it will add additional validation for null everywhere).?

Reply:
`Collections.emptyMap()` always return the same static instance of an empty map.